src/DocumentDB.ChangeFeedProcessor/LeaseManagement/DocumentServiceLeaseStore.cs

```diff
@@ -49,13 +53,18 @@ public async Task MarkInitializedAsync()
         {
             string markerDocId = this.GetStoreMarkerName();
             var containerDocument = new Document { Id = markerDocId };
+            if (this.isPartitionedByLeasePk)
```

Comment:
Let's not special case in this file. Just call request options factory to add PK (when needed).